### DISCUSSION TITLE
Mixed language C/C++ project

### DISCUSSION THREAD

**Discussion by IgorDeepakM** [modular/modular#2760] [Q&A]
I'm trying to find information how to create a mixed language project with Mojo. Mixing with Python is already possible but how about C/C++? I have a code base based on C/C++ and I would like to investigate the possibilities to partially use Mojo in this project.

Currently the tool "mojo build project" only seems to support building the target executable directly (not seeing the intermediate steps). In a mixed language project you would like to compile the Mojo code to object files so that a separate linker step can link the C/C++/Mojo object files together.

1. Does Mojo support mixed language project with a separate linker step?
2. Is there a guide that can show how to create mixed language projects?
3. Does Mojo support compiling to object file format?
4. What dependencies does Mojo require, like libc, libstdc++?

**Comment by billybrickner** [comment]
As a primarily firmware engineer who uses Python for test bench this would solve a lot of issues having to reimplement functionality for test generation/verification. With the idea:

TB from Python to mojo 

Link c objects to test bench

Possibly even just having progressive syntax to do vectorization reasonably I could see some library functions implemented just to accelerate that part of the code.

**Comment by david-ragazzi** [comment]
👍🏻💯

**Comment by rd4com** [comment]
Hello @IgorDeepakM , @billybrickner , @david-ragazzi 

Check out the `▶️ Mojo Community Meeting` of `🗓️ Jul 29, 2024`: 
https://www.youtube.com/watch?v=1T-MBC9k99M&ab_channel=Modular

&nbsp;

`Chris Lattner` talks about Python-Mojo interop and answers questions about C/C++ and mojo.

There is a broad talk on mojo, with questions/answers at the end of the meeting.

&nbsp;

There are also stdlib directions about what need to be done to bring mojo and python closer.

I think you'll find exciting answers in this `video-discussion` :+1:
